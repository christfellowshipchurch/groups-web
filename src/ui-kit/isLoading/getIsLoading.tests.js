import React from 'react';
import renderer from 'react-test-renderer';

// import Providers from '../../Providers';

import { getIsLoading, IsLoadingContext } from '.';

describe('the getIsLoading HOC', () => {
  it('should return false from context', () => {
    const Context = getIsLoading(({ isLoading }) => isLoading.toString());

    const tree = renderer.create(
      <IsLoadingContext.Provider value={false}>
        <Context />
      </IsLoadingContext.Provider>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should return true from context', () => {
    const Context = getIsLoading(({ isLoading }) => isLoading.toString());

    const tree = renderer.create(
      <IsLoadingContext.Provider value>
        <Context />
      </IsLoadingContext.Provider>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should return the isLoading value (true) from the component', () => {
    const Context = getIsLoading(({ isLoading }) => isLoading.toString());

    const tree = renderer.create(
      <IsLoadingContext.Provider>
        <Context isLoading />
      </IsLoadingContext.Provider>
    );

    expect(tree).toMatchSnapshot();
  });
});
