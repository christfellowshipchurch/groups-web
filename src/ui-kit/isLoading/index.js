import React, { useContext } from 'react';

// Create our IsLoadingContext
const IsLoadingContext = React.createContext(false);

/* Safely checks for isLoading to make it easy to build components that respond to either props or
 * context for rendering placeholder/loading state. */
const getIsLoading = (Component) => (props) => {
  // does the component have an isLoading property?
  if (Object.hasOwnProperty.call(props, 'isLoading')) {
    return <Component {...props} />;
  }

  // No? That's fine we'll pull it from context
  const isLoading = useContext(IsLoadingContext);

  return <Component isLoading={isLoading} {...props} />;
};

// Allows a component to accept an `isLoading` prop to pass down into context for it's children.
const withIsLoading = (Component) => ({ isLoading, ...props }) => (
  <IsLoadingContext.Provider value={isLoading}>
    <Component {...props} />
  </IsLoadingContext.Provider>
);

export { getIsLoading, IsLoadingContext, withIsLoading };
