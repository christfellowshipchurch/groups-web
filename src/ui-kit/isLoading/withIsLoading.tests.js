import React from 'react';
import renderer from 'react-test-renderer';

import { getIsLoading, withIsLoading } from '.';

describe('the withIsLoading HOC', () => {
  it('should render', () => {
    const ParentComponentPassingConext = withIsLoading(
      ({ children }) => children
    );

    const tree = renderer.create(
      <ParentComponentPassingConext>
        <h1>Boom</h1>
      </ParentComponentPassingConext>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should pass the isLoading prop (true) to context for children to consume', () => {
    const ChildComponentGettingContext = getIsLoading(({ isLoading }) =>
      isLoading.toString()
    );

    const ParentComponentPassingConext = withIsLoading(() => (
      <ChildComponentGettingContext />
    ));

    const tree = renderer.create(<ParentComponentPassingConext isLoading />);

    expect(tree).toMatchSnapshot();
  });
});
