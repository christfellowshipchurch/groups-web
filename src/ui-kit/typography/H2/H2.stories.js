import React from 'react';
import { storiesOf } from '@storybook/react';

import H2 from '.';

storiesOf('ui-kit/typography/H2', module)
  .add('default', () => (
    <H2>
      {
        '"God is most glorified in us when we are most satisfied in Him" ― John Piper'
      }
    </H2>
  ))
  .add('isLoading', () => (
    <H2 isLoading showLoadingAnimation>
      {
        '"God is most glorified in us when we are most satisfied in Him" ― John Piper'
      }
    </H2>
  ))
  .add('withMargins', () => <H2 withMargins>Padded Heading Big and Long</H2>)
  .add('withMargins + isLoading', () => (
    <H2 isLoading withMargins showLoadingAnimation>
      {
        '"God is most glorified in us when we are most satisfied in Him" ― John Piper'
      }
    </H2>
  ));
