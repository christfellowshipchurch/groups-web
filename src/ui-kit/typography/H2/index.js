import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

import { Line } from '../../Placeholder';

const verticalRhythm = css`
  ${({ padded, theme }) =>
    padded
      ? `${theme.helpers.verticalRhythm(1)} 0 ${theme.helpers.verticalRhythm(
          0.834
        )}`
      : 'initial'};
`;

const StyledH2 = styled.h2.attrs({
  className: 'ui-kit-typography-H2-StyledH2',
})`
  font-size: 2.25rem;
  font-weight: 900;
  line-height: ${({ theme }) => theme.helpers.verticalRhythm(2.25, 1.15)};
  margin: ${verticalRhythm};
`;

const Placeholder = styled(Line).attrs({
  className: 'ui-kit-H1-Placeholder',
})`
  height: 2.25rem; /* Should match .h2font-size */
  margin: ${({ padded, theme }) =>
    // if padded...
    padded
      ? // return rhythm so placeholder matches text position to prevent jitter/jank
        verticalRhythm
      : // else default to...
        `calc(${theme.helpers.verticalRhythm(
          2.25,
          1.15
        )} - 2.25rem) 0`}; /* line-height - font-size = space between lines */
`;

const H2 = ({ children, isLoading, padded, ...props }) => (
  <Placeholder isLoading={isLoading} padded={padded} showLoadingAnimation>
    <StyledH2 padded={padded} {...props}>
      {children}
    </StyledH2>
  </Placeholder>
);

H2.defaultProps = {
  isLoading: false,
};

H2.propTypes = {
  children: PropTypes.node.isRequired,
  isLoading: PropTypes.bool,
  padded: PropTypes.bool,
};

H2.displayName = 'ui-kit.typography.H2';

export default H2;
