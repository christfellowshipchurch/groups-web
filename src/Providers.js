import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { defaultTheme } from './ui-kit/theme';

const Providers = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
);

Providers.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Providers;
