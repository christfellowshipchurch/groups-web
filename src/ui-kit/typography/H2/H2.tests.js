import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Providers from '../../../Providers';
import { IsLoadingContext } from '../../isLoading';

import H2 from '.';

describe('the H2 component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <H2>Default H2 text</H2>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept and render passed in styles', () => {
    const salmonText = { color: 'salmon' };
    const tree = renderer.create(
      <Providers>
        <H2 style={salmonText}>Salmon text</H2>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with vertical rhythm (padded)', () => {
    const tree = renderer.create(
      <Providers>
        <H2 padded>H2 text with vertical rhythm styled via margin.</H2>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state', () => {
    const tree = renderer.create(
      <Providers>
        <H2 isLoading>Default H2 text</H2>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state via context', () => {
    const tree = renderer.create(
      <Providers>
        <IsLoadingContext.Provider value>
          <H2 isLoading>Default H2 text</H2>
        </IsLoadingContext.Provider>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state with vertical rhythm (padded)', () => {
    const tree = renderer.create(
      <Providers>
        <H2 isLoading padded>
          Default H2 text
        </H2>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
