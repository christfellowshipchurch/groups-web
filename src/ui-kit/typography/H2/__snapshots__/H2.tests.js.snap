// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the H2 component should accept and render passed in styles 1`] = `
<h2
  className="r3v0jj-0 cdBPqM ui-kit-typography-H2-StyledH2"
  style={
    Object {
      "color": "salmon",
    }
  }
>
  Salmon text
</h2>
`;

exports[`the H2 component should render 1`] = `
<h2
  className="r3v0jj-0 cdBPqM ui-kit-typography-H2-StyledH2"
>
  Default H2 text
</h2>
`;

exports[`the H2 component should render a loading state 1`] = `
<mark
  className="ra2rjz-0 gcMOrm show-loading-animation r3v0jj-1 kqMxnS ui-kit-H1-Placeholder ui-kit.Placeholder.Line custom-placeholder"
  width="100%"
/>
`;

exports[`the H2 component should render a loading state with vertical rhythm (padded) 1`] = `
<mark
  className="ra2rjz-0 gcMOrm show-loading-animation r3v0jj-1 iTfMnP ui-kit-H1-Placeholder ui-kit.Placeholder.Line custom-placeholder"
  width="100%"
/>
`;

exports[`the H2 component should render with vertical rhythm (padded) 1`] = `
<h2
  className="r3v0jj-0 gguShV ui-kit-typography-H2-StyledH2"
>
  H2 text with vertical rhythm styled via margin.
</h2>
`;
